src/socket_base.cpp

```diff
@@ -1630,7 +1630,7 @@ void zmq::socket_base_t::pipe_terminated (pipe_t *pipe_)
 void zmq::socket_base_t::extract_flags (msg_t *msg_)
 {
     //  Test whether routing_id flag is valid for this socket type.
-    if (unlikely (msg_->flags () & msg_t::routing_id))
+    if (unlikely ((msg_->flags () & msg_t::routing_id) == msg_t::routing_id))
```

Comment:
also not sure this is required? if the routing_id bit is set, then the result cannot be anything else that routing_id, which already evaluates as true

Reply:
no its not required